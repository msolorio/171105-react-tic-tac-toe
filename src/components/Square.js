import React from 'react';
import PropTypes from 'prop-types';

export default function Square(props) {

  function handleSquareClick() {
    props.onSquareClick(props.index);
  }

  return (
    <div className={`Square Square-${props.index}`}
      onClick={handleSquareClick}>
      {props.squareVal}
    </div>
  );
}

Square.propTypes = {
  index: PropTypes.number.isRequired,
  squareVal: PropTypes.string.isRequired,
  onSquareClick: PropTypes.func.isRequired
};
