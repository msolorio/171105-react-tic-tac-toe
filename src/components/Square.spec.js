import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Square from './Square';

const squareProps = {
  index: 0,
  squareVal: '',
  onSquareClick: jest.fn()
};

describe('Square', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(
      <Square {...squareProps} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render squareVal props as text within each square', () => {
    const squareVals = ['X', 'O', ''];

    squareVals.forEach((val) => {
      const testProps = {
        index: 3,
        squareVal: val,
        onSquareClick: jest.fn()
      };

      const square = shallow(<Square {...testProps} />);
      expect(square.text()).toBe(testProps.squareVal);
    });
  });

  it('should call props.onSquareClick with the square index on square click', () => {
    for (let i=0; i<9; i++) {
      const testProps = {
        index: i,
        squareVal: '',
        onSquareClick: jest.fn()
      };

      const square = shallow(<Square {...testProps} />);
      square.simulate('click');
      expect(testProps.onSquareClick.mock.calls.length).toBe(1);
      /*
      * expect first argument of first call of onSquareClick
      * to be index value passed as props
      */
      expect(testProps.onSquareClick.mock.calls[0][0]).toBe(i);
    }
  });
});
