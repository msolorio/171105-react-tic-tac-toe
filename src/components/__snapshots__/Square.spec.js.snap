// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Square should match snapshot 1`] = `
<div
  className="Square Square-0"
  onClick={[Function]}
>
  
</div>
`;
